File: core/src/main/java/org/keycloak/representations/idm/ConfigPropertyRepresentation.java
package org.keycloak.representations.idm;

/**
* @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
* @version $Revision: 1 $
*/
public class ConfigPropertyRepresentation {
    protected String name;
    protected String label;
    protected String helpText;
    protected String type;
    protected Object defaultValue;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Object getDefaultValue() {
        return defaultValue;
    }

    public void setDefaultValue(Object defaultValue) {
        this.defaultValue = defaultValue;
    }

    public String getHelpText() {
        return helpText;
    }

    public void setHelpText(String helpText) {
        this.helpText = helpText;
    }
}

File: js/libs/keycloak-admin-client/src/defs/configPropertyRepresentation.ts
/**
 * https://www.keycloak.org/docs-api/11.0/rest-api/index.html#_configpropertyrepresentation
 */
export interface ConfigPropertyRepresentation {
  name?: string;
  label?: string;
  helpText?: string;
  type?: string;
  defaultValue?: object;
  options?: string[];
  secret?: boolean;
  required?: boolean;
}
